pkg/state/factory.go

```diff
@@ -8,12 +12,28 @@ const (
 	GroupConfigsDir = "groups"
 )
 
-// NewRunConfigStore creates a store for run configuration state
+// NewRunConfigStore creates a local store for run configuration state
 func NewRunConfigStore(appName string) (Store, error) {
+	return NewRunConfigStoreWithDetector(appName, nil)
+}
+
+// NewRunConfigStoreWithDetector creates a local store (detector parameter ignored)
+func NewRunConfigStoreWithDetector(appName string, _ any) (Store, error) {
+	if runtime.IsKubernetesRuntime() {
+		return nil, nil // No store needed in Kubernetes environments
```

Comment:
This will segfault anything which expects this method to return an actual instance. Should we have an interface with a no-op implementation.

Reply:
it was on previous prs. So let me add it back and we discuss